tests/backends/data/test_s3.py

```diff
@@ -0,0 +1,568 @@
+    `FileExistsError`.
+    """
+    # Regions outside of us-east-1 require the appropriate LocationConstraint
+    s3_client = boto3.client("s3", region_name="us-east-1")
+    # Create a valid bucket in Moto's 'virtual' AWS account
+    bucket_name = "bucket_name"
+    s3_client.create_bucket(Bucket=bucket_name)
+
+    body = b"some contents in the body"
+
+    s3_client.put_object(
+        Bucket=bucket_name,
+        Key="2022-09-29.gz",
+        Body=body,
+    )
+
+    object_name = "2022-09-29.gz"
+    some_content = b"some contents in the stream file to upload"
+
+    caplog.set_level(logging.ERROR)
+
+    with pytest.raises(BackendException):
+        s3 = s3_backend()
+        s3.clean_history(lambda *_: True)
+        s3.write(data=some_content, target=object_name, operation_type=operation_type)
+    logger_name = "ralph.backends.data.s3"
+    msg = (
+        f"{object_name} already exists and overwrite is not allowed for operation"
+        f" {operation_type if operation_type is not None else BaseOperationType.CREATE}"
+    )
+    assert caplog.record_tuples == [(logger_name, logging.ERROR, msg)]
+    assert s3.history == []
+
+
+@pytest.mark.parametrize(
+    "operation_type",
+    [BaseOperationType.APPEND, BaseOperationType.DELETE],
+)
+def test_backends_data_s3_data_backend_write_method_with_append_or_delete_operation(
+    s3_backend, operation_type
+):
+    """Test the `S3DataBackend.write` method, given an `APPEND`
+    `operation_type`, should raise a `BackendParameterException`.
+    """
+    # pylint: disable=invalid-name
+    backend = s3_backend()
+    with pytest.raises(
+        BackendParameterException,
+        match=f"{operation_type.name} operation_type is not allowed.",
+    ):
+        backend.write(data=[b"foo"], operation_type=operation_type)
+
+
+@pytest.mark.parametrize(
+    "operation_type",
+    [BaseOperationType.CREATE, BaseOperationType.INDEX],
+)
+@mock_s3
+def test_backends_data_s3_write_method_with_create_index_operation(
+    operation_type, s3_backend, monkeypatch
+):  # pylint: disable=invalid-name
+    """Test the `S3DataBackend.write` method, given a target matching an
+    existing object and a `CREATE` or `INDEX` `operation_type`, should add
+    an entry to the History.
+    """
+    # Regions outside of us-east-1 require the appropriate LocationConstraint
+    s3_client = boto3.client("s3", region_name="us-east-1")
+    # Create a valid bucket in Moto's 'virtual' AWS account
+    bucket_name = "bucket_name"
+    s3_client.create_bucket(Bucket=bucket_name)
+
+    freezed_now = datetime.datetime.now(tz=datetime.timezone.utc).isoformat()
+    monkeypatch.setattr("ralph.backends.data.s3.now", lambda: freezed_now)
+
+    object_name = "new-archive.gz"
+    some_content = b"some contents in the stream file to upload"
+
+    s3 = s3_backend()
+    s3.clean_history(lambda *_: True)
+
+    response = s3.write(
+        data=[some_content],
```

Comment:
```suggestion
        data=[some_content, some_content, some_content],
```
Could you try to test the write method with multiple chunks?
Isn't the `put_object` method overwriting the last chunk on each iteration?
If that's the case, maybe we could use the `upload_fileobj` method instead (converting `data` to a stream if it's an Iterable)?

Reply:
Dang, I don't know what I was doing, but you are right, thank you for spotting this! Reverting to using `upload_fileobj`, with a conversion from an iterable to a file-like object. Let me know what you think!